agent/hcp/telemetry/otel_exporter_test.go

```diff
@@ -137,3 +192,15 @@ func mutateMetrics(m []metricdata.ScopeMetrics) *metricdata.ResourceMetrics {
 		ScopeMetrics: m,
 	}
 }
+
+func performExporterOperation(exp metric.Exporter, operation string) {
```

Comment:
This was the nicest way I could think of doing this, but happy to hear other ideas 😓 

The alternative was to export, force flush and shutdown in a row, but I'd have to re-init the exporter for the exportFailure with a bad client... so it wasn't looking better.

Reply:
My personal opinion/approach would be to make `operation` a field on the test struct. It's a bit tricky where the operation here is the name of the test, means having to hop back and forth to see what's tested. If you had this switch statement within the test body I think that's cleaner (albeit a longer test)

Reply:
Updated! Looks much better, let me know if this is what you were thinking
- Added operation as field name in test struct
- Moved switch to within the test
- Clarifies test names (so they're not repeats of the operation)